src/relations/db.py

```diff
@@ -289,22 +289,27 @@ def update_port(self, relation: Relation, port: str):
             logger.warning("relation not fully initialised - skipping port update")
             return
 
-        dbconnstr = pgb.parse_dict_to_kv_string(
-            {
-                "host": self.charm.unit_pod_hostname,
-                "dbname": database,
-                "port": port,
-                "user": user,
-                "password": password,
-                "fallback_application_name": self.get_external_app(relation).name,
-            }
-        )
+        master_dbconnstr = {
+            "host": self.charm.peers.leader_hostname,
```

Comment:
Sorry I don't know enough about PGB & PG, is it gaurenteed that the leader is the master?

Also nit at some point you'll want to replace master with something else, see [this convo](https://chat.canonical.com/canonical/pl/kb7k7pnhbifyb8mqoubg8zb1ky), feel free to do this throughout in a separate PR.

Reply:
For pgbouncer, the leader/master distinction is irrelevant. Unfortunately the relation interface relies on this outdated terminology, so we're forced to use it